platform/components/tekton/tasks/patch_trivy.yml

```diff
@@ -1,8 +1,19 @@
+- op: add
+  path: /spec/steps/0
+  value:
+    name: get-latest-cve-findings
+    image: $(params.TRIVY_IMAGE)
+    script: |
+      TRIVY_TEMP_DIR=$(mktemp -d)
+      trivy --cache-dir $TRIVY_TEMP_DIR image --download-db-only
```

Comment:
Is this step subject to the same Aqua rate limits? Unless the output of this step is cached between runs, we'll execute the non-airgapped command on every pipeline that calls the `trivy-scanner` task. If it is cached between runs, we'll need some logic to know when to refresh the cache.

Reply:
Hi, @developer-guy brought me here as I'm a maintainer of Trivy. First of all, I'm sorry, but I'm not familiar with this project. I may be saying something wrong.

The Trivy database is distributed on GitHub. `trivy image --download-db-only` just downloads this database from GitHub. But you can use any tool like [ORAS](https://oras.land/). You may need to be concerned about rate limiting on GitHub.
https://github.com/aquasecurity/trivy-db/pkgs/container/trivy-db

`trivy image --download-db-only` downloads the database to `$CACHE_DIR/trivy/db` and skips downloading it until the next update. `$CACHE_DIR` depends on the environment. It may be `$HOME/.cache` on Linux.

And `NextUpdate` is stored in `metadata.json` under the cache directory.

```
$ jq . $CACHE_DIR/trivy/db/metadata.json
{
  "Version": 2,
  "NextUpdate": "2023-01-12T12:07:03.840852271Z",
  "UpdatedAt": "2023-01-12T06:07:03.840852771Z",
  "DownloadedAt": "2023-01-12T11:44:20.307947Z"
}
```

It needs to update the database every 6 hours, but you can skip it with `--skip-db-update` if you want.

Hopefully, my answer helps.

Reply:
Hello @knqyf263, Thank you for the feedback on this comment. Correct me if I am wrong but doesn't the downloaded database cache need to be stored in persistent storage for it to be used for other pipeline runs? Tekton uses ephemeral environments so the cache would be deleted after each run.

Reply:
> doesn't the downloaded database cache need to be stored in persistent storage for it to be used for other pipeline runs?

Yes, the cache directory needs to be stored in persistent storage.

Reply:
@developer-guy so I see there is a PVC in place for the taskrun but to @sudo-bmitch point seems like there should be logic to check if the cache is there and update it after 6 hours based on the above comments.

Reply:
AFAIK, trivy will do that for us by looking at the `NextUpdate` field of the metadata information, am I wrong @knqyf263 ?

Reply:
>AFAIK, trivy will do that for us by looking at the NextUpdate field of the metadata information

Yes, `trivy --download-db-only` checks if the download should be updated or not, then downloads it if needed.

Reply:
I missed that we have a PVC buried in there. I'll resolve this and open a separate review for an issue I'm seeing.